tools/devops/automation/templates/packages/build.yml

```diff
@@ -0,0 +1,454 @@
+    provisioning_extra_args: '-vvvv'
+  timeoutInMinutes: 30
+  enabled: false
+
+- bash: |
+    make -C $(Build.SourcesDirectory)/xamarin-macios/tools/devops build-provisioning.csx
+  displayName: 'Generate provisionator files.'
+
+- task: xamops.azdevex.provisionator-task.provisionator@1
+  displayName: 'Provision Products & Frameworks'
+  inputs:
+    provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/build-provisioning.csx
+    provisioning_extra_args: '-vvvv'
+  timeoutInMinutes: 250
+
+- task: xamops.azdevex.provisionator-task.provisionator@1
+  displayName: 'Provision .NET Core'
+  inputs:
+    provisioning_script: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/provision-dotnetcore.csx
+    provisioning_extra_args: '-vvvv'
+  timeoutInMinutes: 30
+
+- powershell: |
+    if ([IO.Directory]::Exists('/Developer/MonoTouch')) {
+      Write-Host "Deleting /Developer/MonoTouch"
+      bash -c "sudo rm -Rf /Developer/MonoTouch"
+    }
+
+    if ([IO.Directory]::Exists('/Library/Frameworks/Xamarin.iOS.framework')) {
+      Write-Host "Deleting /Library/Frameworks/Xamarin.iOS.framework"
+      bash -c "sudo rm -Rf /Library/Frameworks/Xamarin.iOS.framework"
+    }
+
+    if ([IO.Directory]::Exists('/Library/Frameworks/Xamarin.Mac.framework')) {
+      Write-Host "Deleting /Library/Frameworks/Xamarin.Mac.framework"
+      bash -c "sudo rm -Rf /Library/Frameworks/Xamarin.Mac.framework"
+    }
+  displayName: 'Delete library folders'
+  timeoutInMinutes: 5
+
+- bash:
+    set -x
+    set -e
+    rm -Rvf $(Build.SourcesDirectory)/package
+    time make -C $(Build.SourcesDirectory)/xamarin-macios/ git-clean-all
+  displayName: 'Clear results directory'
+  timeoutInMinutes: 5
+
+# Use the env variables that were set by the label parsing in the configure step
+# print some useful logging to allow to know what is going on AND allow make some
+# choices, there are labels that contradict each other (skip-package vs build-packages)
+# we use warnings for those case we are not sure about.
+- pwsh: |
+    # we have a number of scripts that require to be executed from the top of the src, rather
+    # than keeping track of the location of the script, we create two env vars that can be used to
+    # get to the top
+    $xamTop = "$(Build.SourcesDirectory)/xamarin-macios/"
+    Write-Host "##vso[task.setvariable variable=XAM_TOP]$xamTop"
+
+    $maccoreTop = "$(Build.SourcesDirectory)/maccore/"
+    Write-Host "##vso[task.setvariable variable=MACCORE_TOP]$maccoreTop"
+
+    $buildReason = "$(Build.Reason)"
+    $buildSourceBranchName = "$(Build.SourceBranchName)"
+
+    # decide if we are dealing with a PR or a re-triggered PR or a build from
+    # a branch in origin
+    
+    if ($buildReason -eq "PullRequest" -or (($buildReason -eq "Manual") -and ($buildSourceBranchName -eq "merge")) ) {
+      Write-Host '##vso[task.setvariable variable=IsPR;isOutput=true]False'
+
+      if ($Env:BuildPackage -eq "True") {
```

Comment:
Labels can be set for all builds, not only PR builds, so this logic should always be executed.